### PR TITLE
Fix statement disappear issue

### PR DESCRIPTION
**Pull request by 4p00rv**
Related [trello card](https://trello.com/c/Ws6K98yT/1508-purchase-statement-is-to-fast-close).

### CODE REVIEW

**Comment by raunakkathuria** [comment]
@4p00rv can you provide detail about fix i.e. regarding changes, issue details we have in trello

File: src/javascript/binary/pages/trade/event.js [hunk 1]
```diff
@@ -172,17 +172,28 @@ var TradingEvents = (function () {
...
+            sessionStorage.setItem('duration_amount',e.target.value);
+            Durations.select_amount(e.target.value);
+            processPriceRequest();
+            submitForm(document.getElementById('websocket_form'));
+        }
+        var durationAmountElement = document.getElementById('duration_amount'),
+            flagForEvent = 0;          // For triggering one of the two events.
         if (durationAmountElement) {
             // jquery needed for datepicker
-            $('#duration_amount').on('input change', debounce(function (e) {
```
Comment (4p00rv) [inline src/javascript/binary/pages/trade/event.js]: @raunakkathuria Sure. Earlier both the events were triggered one after another which invoked `processPriceRequest` twice. The issue was when the user used to change the duration using the text-box and right after that purchased the contract, the statement box used close automatically within seconds. So to trigger only one of the two events I used a flag `flagForEvent` which avoids this behavior.

File: src/javascript/binary/pages/trade/event.js [hunk 2]
```diff
@@ -172,17 +172,28 @@ var TradingEvents = (function () {
...
+            if (e.target.value % 1 !== 0 ) {
+                e.target.value = Math.floor(e.target.value);
+            }
+            sessionStorage.setItem('duration_amount',e.target.value);
+            Durations.select_amount(e.target.value);
+            processPriceRequest();
+            submitForm(document.getElementById('websocket_form'));
+        }
+        var durationAmountElement = document.getElementById('duration_amount'),
+            flagForEvent = 0;          // For triggering one of the two events.
```
Comment (4p00rv) [inline src/javascript/binary/pages/trade/event.js]: This is the flag.

File: src/javascript/binary/pages/trade/event.js [hunk 3]
```diff
@@ -172,17 +172,28 @@ var TradingEvents = (function () {
         /*
          * bind event to change in duration amount, request new price
          */
-        var durationAmountElement = document.getElementById('duration_amount');
+        function triggerOnDurationChange(e){
+            if (e.target.value % 1 !== 0 ) {
+                e.target.value = Math.floor(e.target.value);
+            }
```
Comment (qingweibinary) [inline src/javascript/binary/pages/trade/event.js]: Suggest to use 
```
var roundedValue = Math.floor(e.target.value);

// use roundedValue onwards
```
this make it clearer we need an Integer, and reduce side effect

File: src/javascript/binary/pages/trade/event.js (hunk 2 above)
Comment (qingweibinary) [inline src/javascript/binary/pages/trade/event.js]: Can the name be more clear ? (I am not sure what it means exactly)
```
inputEventTriggered = false;
``` 

if it only have 2 values, use a boolean

Comment (4p00rv) [inline src/javascript/binary/pages/trade/event.js]: noted!

File: src/javascript/binary/pages/trade/event.js (hunk 3 above)
Comment (4p00rv) [inline src/javascript/binary/pages/trade/event.js]: It was older code though, but I what we're doing here is changing the value in the text box to integer by replacing the existing value i.e `e.target.value`

Comment (qingweibinary) [inline src/javascript/binary/pages/trade/event.js]: I see, that's fine then ;)

**Pull request merged**